core/drivers/lpc_uart.c

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2022, HiSilicon Limited
+ */
+
+#include <assert.h>
+#include <drivers/lpc_uart.h>
+#include <io.h>
+#include <keep.h>
+#include <mm/core_mmu.h>
+#include <util.h>
+
+static paddr_t chip_to_base(struct serial_chip *chip)
+{
+	struct lpc_uart_data *pd =
+		container_of(chip, struct lpc_uart_data, chip);
+
+	return io_pa_or_va(&pd->base, LPC_SIZE);
+}
+
+static void lpc_byte_read(paddr_t addr, unsigned char *data)
+{
+	unsigned int cnt = 0;
+	unsigned int status;
+
+	io_write32((LPC_CMD_REG_OFFSET) + addr, 0x8);
```

Comment:
Please remove the useless parentheses: `io_write32(LPC_CMD_REG_OFFSET + addr, 0x8);`
And in other places too.